Update FR locales

eramdam opened the pull request:
- [x] Add locales for the new profile view
- [x] Add locales for the new pin feature
- [x] Fix missing locales in `doorkeeper.fr.yml` and clean un-used ones
- [x] Fix a few typos in the locales

ykzts commented:
cc @aaribaud @Aldarone @Angristan @BoFFire @bzg @cgxxx @cyweo @Eychics @goofy-bz @Jnsll @kklleemm @Komic @nclm @ProgVal @SansPseudoFix @Sylvhem @tcitworld @trebmuh

Aldarone commented on app/javascript/mastodon/locales/fr.json:
```diff
@@ -43,7 +43,7 @@
   "column_subheading.navigation": "Navigation",
   "column_subheading.settings": "Paramètres",
-  "compose_form.lock_disclaimer": "Votre compte n’est pas {locked}. Tout le monde peut vous suivre et voir vos pouets restreints.",
+  "compose_form.lock_disclaimer": "Votre compte n’est pas {locked}. Tout le monde peut vous suivre et voir vos statuts privés.",
```
Why did you replace « pouet » with « statuts » ? 😭

Aldarone commented on config/locales/fr.yml:
```diff
@@ -404,16 +426,23 @@ fr:
     over_character_limit: limite de caractères dépassée de %{max} caractères
+    pin_errors:
+      ownership: Vous ne pouvez pas épingler un statut ne vous appartenant pas
+      private: Non-public toot cannot be pinned
```
You forgot one here 😉

SansPseudoFix commented on config/locales/fr.yml:
```diff
@@ -212,7 +226,13 @@ fr:
     invalid_url: L’URL fournie est invalide
+    regenerate_token: Regénérer le jeton d'accès
+    token_regenerated: Jeton d'accès regénéré avec succès
+    warning: Soyez prudent⋅e avec ces données. Ne les partagez pas!
```
Typo: in french, there is a space before (and after) `!`

aaribaud replied:
More precisely, in French the exclamation mark requires a thin non breakable space ('espace fine') before and a plain (breakable) space after (as do interrogation point, colon, semicolon and some others).

SansPseudoFix commented on app/javascript/mastodon/locales/fr.json:
```diff
@@ -180,7 +180,7 @@
   "status.show_more": "Déplier",
   "status.unmute_conversation": "Ne plus masquer la conversation",
-  "status.unpin": "Unpin from profile",
+  "status.unpin": "Désépingler du profil",
```
I understand why, but I don't think `désépingler` is a good idea. Why not "relâcher" like Firefox does?

Komic replied:
I think "_Relâcher_" is too vague. "_Désépingler_" is a valid word _and_ more precise so why not~

aaribaud replied:
I'm not too fond of "désépingler", which sounds like it was made up for the occasion -- although it does exist in French as a construction based on "épingler", its meaning is "remove pins from sth" rather than "free sth from the the pins which held it".

How about "Retirer"? Its common meaning include "remove a publication from display".

eramdam replied:
I went with "désépingler" because this is how it was translated in French in Twitter's UI 
![image](https://user-images.githubusercontent.com/1409924/29752169-f943a8c0-8b0d-11e7-84ef-a60d0ffddfc5.png)

As much as the word seems "weird" I agree with @Komic that it's more precise anyway, and will feel familiar instantly to those who used Twitter before coming to Mastodon since it's basically the same concept/feature.

aaribaud replied:
I'm not too fond of a translation choice that would be summed up as "let's use twitter-specific unusual French rather than plain simple French". In translation as in many other domains, simplicity is important.

Komic commented:
I personally don't really like how "_pouet_" sounds in French but IMO it should be kept.

trebmuh commented:
I agree with others' comments.
Apart from those details, good job and thanks @eramdam !

bzg commented:
> How about "Retirer"? Its common meaning include "remove a publication from display".

A strong +1 for "retirer".

Komic commented:
sure, whatever

eramdam commented:
@Aldarone @Komic @tcitworld The "pouet" was removed from basically everywhere else in the locales but one string, I can go and put it back everywhere else, but, like, I'm not sure when and how it was removed in the first place.

Aldarone commented:
@eramdam It seems that in the english strings there is a mix of « post » and « status » which saddens me a bit more. I won't object sticking close to the original strings though.

eramdam commented:
@Aldarone yeah, it doesn't help. I'd say I can remove the commit on this specific PR and we can do/discuss the change (or not) later on, because it's quite a big chunk of strings to update.

kklleemm commented on config/locales/fr.yml:
```diff
@@ -404,16 +426,23 @@ fr:
+      ownership: Vous ne pouvez pas épingler un statut ne vous appartenant pas
+      private: Non-public toot cannot be pinned
+      private: Les statuts non-publics ne peuvent pas être épinglés
+      reblog: Un partage ne peut être épinglé
```
What about "ne peut pas" ? (just like line 444 above)

eramdam commented:
@tcitworld @Aldarone @kklleemm I fixed the things you pointed out.

Now about the whole `pouet` situation: 
1. The FR locales already had a mix of `statuts` and `pouets`, I barely removed the last `pouet` (outside of the "send" button in the composer) for the sake of uniformity
2. The EN locales also have a mix of `toot`, `posts` and `status`, and I'm not sure what's the logic to follow is supposed to be
3. I put back the `pouet` that I removed with this PR
4. I'm more than happy to discuss a way to improve this whole thing w/ everybody involved, but I think it deserves its own PR. I think we had to find kind of a "logic" to follow, to decide when to use `statut` or when to use `pouets`, because some parts of the UI would be weird with `pouet` and others might be completely fine with it.

Gargron commented:
>The EN locales also have a mix of toot, posts and status, and I'm not sure what's the logic to follow is supposed to be

I wish I could answer that, but it's mostly just a chronological progression of the project!

Gargron commented:
Is this good to merge?
